lib/steep/type_inference/logic_type_interpreter.rb

```diff
@@ -405,6 +405,38 @@ def evaluate_method_call(env:, type:, receiver:, arguments:)
               truthy_result.update_type { FALSE }
             ]
           end
+
+        when AST::Types::Logic::Guard
+          if receiver
+            receiver_type = factory.deep_expand_alias(typing.type_of(node: receiver)) || raise
+
+            # TODO: Make the type absolute
```

Comment:
I'd like to know the best way to absolute the type in the type guard annotation. `factory#absolute_type_name`?